### PR TITLE
YR/PAN-OS-Policy-Optimizer-V10.2.0/XSUP-25940/CIAC-7392

### PR DESCRIPTION
**Pull request by RosenbergYehuda**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/XSUP-25940, https://jira-hq.paloaltonetworks.local/browse/CIAC-7392 

## Description

- Changed the token generator to use sha256 for newer Pan-os versions.

- Changed the test module to be more sensitive and give a better indication.

- Fixed an issue in 3 command that did not work for panorama instance.
this should be probably fixed in all versions,
but for now we'll just fix it for version 10.2.0 and above since that's the version we have access to.
The commands are:
  - pan-os-po-get-stats.
  - pan-os-po-no-apps
  - pan-os-po-unused-apps

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/PANOSPolicyOptimizer/Integrations/PANOSPolicyOptimizer/PANOSPolicyOptimizer.yml [hunk 1]
```diff
@@ -39,6 +39,14 @@ script:
   commands:
   - description: Gets the Policy Optimizer statistics.
     name: pan-os-po-get-stats
+    arguments:
+    - auto: PREDEFINED
+      description: Whether to get pre-rules statistics, post-rules statistics. 'pre' for pre rules, 'post' for post-rules, only for panorama instances.
```
Comment (ShirleyDenkberg) [inline Packs/PANOSPolicyOptimizer/Integrations/PANOSPolicyOptimizer/PANOSPolicyOptimizer.yml:44]:
```suggestion
      description: Whether to get pre-rules statistics or post-rules statistics. 'pre' for pre rules, 'post' for post-rules. Only for Panorama instances.
```

File: Packs/PANOSPolicyOptimizer/Integrations/PANOSPolicyOptimizer/PANOSPolicyOptimizer.yml [hunk 2]
```diff
@@ -57,12 +65,28 @@ script:
       type: Number
   - description: Shows all security policies with no apps specified.
     name: pan-os-po-no-apps
+    arguments:
+    - auto: PREDEFINED
+      description: Whether to get pre-rules with no apps or post-rules with no apps, 'pre' for pre rules, 'post' for post-rules, only for panorama instances.
```
Comment (ShirleyDenkberg) [inline Packs/PANOSPolicyOptimizer/Integrations/PANOSPolicyOptimizer/PANOSPolicyOptimizer.yml:70]:
```suggestion
      description: Whether to get pre-rules with no apps or post-rules with no apps. 'pre' for pre rules, 'post' for post-rules. Only for Panorama instances.
```

File: Packs/PANOSPolicyOptimizer/Integrations/PANOSPolicyOptimizer/PANOSPolicyOptimizer.yml [hunk 3]
```diff
@@ -57,12 +65,28 @@ script:
...
+      - post
     outputs:
     - contextPath: PanOS.PolicyOptimizer.NoApps
       description: Contains information about the rules that have no apps specified. For example, Source and Destination.
       type: Unknown
   - description: Gets the unused apps.
     name: pan-os-po-unused-apps
+    arguments:
+    - auto: PREDEFINED
+      description: Whether to get pre-rules unused apps, or post-rules unused apps. 'pre' for pre rules, 'post' for post-rules, only for panorama instances.
```
Comment (ShirleyDenkberg) [inline Packs/PANOSPolicyOptimizer/Integrations/PANOSPolicyOptimizer/PANOSPolicyOptimizer.yml:84]:
```suggestion
      description: Whether to get pre-rules unused apps or post-rules unused apps. 'pre' for pre rules, 'post' for post-rules. Only for Panorama instances.
```

File: Packs/PANOSPolicyOptimizer/Integrations/PANOSPolicyOptimizer/README.md [hunk 4]
```diff
@@ -33,7 +32,9 @@ Gets the Policy Optimizer statistics.
 `pan-os-po-get-stats`
 #### Input
 
-There are no input arguments for this command.
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| position | Whether to get pre-rules statistics, post-rules statistics. 'pre' for pre rules, 'post' for post-rules, only for panorama instances. Possible values are: pre, post. Default is pre. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/PANOSPolicyOptimizer/Integrations/PANOSPolicyOptimizer/README.md:37]:
```suggestion
| position | Whether to get pre-rules statistics or post-rules statistics. 'pre' for pre rules, 'post' for post-rules. Only for Panorama instances. Possible values are: pre, post. Default is pre. | Optional | 
```

File: Packs/PANOSPolicyOptimizer/Integrations/PANOSPolicyOptimizer/README.md [hunk 5]
```diff
@@ -88,7 +89,9 @@ Shows all security policies with no apps specified.
 `pan-os-po-no-apps`
 #### Input
 
-There are no input arguments for this command.
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| position | Whether to get pre-rules with no apps or post-rules with no apps. 'pre' for pre rules, 'post' for post-rules, only for panorama instances. Possible values are: pre, post. Default is pre. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/PANOSPolicyOptimizer/Integrations/PANOSPolicyOptimizer/README.md:94]:
```suggestion
| position | Whether to get pre-rules with no apps or post-rules with no apps. 'pre' for pre rules, 'post' for post-rules. Only for Panorama instances. Possible values are: pre, post. Default is pre. | Optional | 
```

File: Packs/PANOSPolicyOptimizer/Integrations/PANOSPolicyOptimizer/README.md [hunk 6]
```diff
@@ -193,7 +198,9 @@ Gets the unused apps.
 `pan-os-po-unused-apps`
 #### Input
 
-There are no input arguments for this command.
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| position | Whether to get pre-rules unused apps, or post-rules unused apps. 'pre' for pre rules, 'post' for post-rules, only for panorama instances. Possible values are: pre, post. Default is pre. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/PANOSPolicyOptimizer/Integrations/PANOSPolicyOptimizer/README.md:203]:
```suggestion
| position | Whether to get pre-rules unused apps or post-rules unused apps. 'pre' for pre rules, 'post' for post-rules. Only for Panorama instances. Possible values are: pre, post. Default is pre. | Optional | 
```

**Comment by ShirleyDenkberg** [comment]
@yuvalbenshalom Doc review completed.

File: Packs/PANOSPolicyOptimizer/ReleaseNotes/1_1_9.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### PAN-OS Policy Optimizer (Beta)
+
+- You can now use the PAN-OS Policy Optimizer (Beta) integration with version 10.2.0 of PAN-OS.
```
Comment (yuvalbenshalom) [inline Packs/PANOSPolicyOptimizer/ReleaseNotes/1_1_9.md:6]:
```suggestion
- Added support for version 10.2.0 of PAN-OS.```

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
see my comments

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**